### DISCUSSION TITLE
Error: Encrypted private key is authenticated using an insecure two-byte hash

### DISCUSSION THREAD

**Discussion by martgil** [openpgpjs/openpgpjs#1525] [Q&A] [answered]
Greetings to all!

I'm humbling want to know what causes the error (1) "Encrypted private key is authenticated using an insecure two-byte hash"? 
(2) Is the s2k usage similar to key usage flags? If so, is it possible to dump the s2k information using a tool eg. hokey pgpdump gpg2?

Many thanks,
Mart

**Comment by larabr** [comment, marked as answer]
Hi @martgil ,
the error is due to a legacy (and insecure) key encryption mechanism; see also https://github.com/openpgpjs/openpgpjs/discussions/1212#discussioncomment-354121.

S2K usage has a different purpose than key usage flags, as it's only needed to decrypt the key. You should still be able to dump the s2k info with many tools. I think GPG should also allow importing/decrypting the key (might require using some specific flag), and then you can re-export it with the same passphrase, to have it in the newer format.

**Reply by martgil** [reply]
Hello @larabr,

Thanks a lot for your time and efforts in answering the question!

> I think GPG should also allow importing/decrypting the key (might require using some specific flag), and then you can re-export it with the same passphrase, to have it in the newer format.

Thanks! - We will give it a try.

**Reply by SudhirkumarRamani** [reply]
@martgil were you able to fix this issue?
